src/nsidc/iceflow/data/atm1b.py

```diff
@@ -286,42 +310,122 @@ def _qfit_file_header(filepath: Path) -> str:
     raise RuntimeError(err)
 
 
-def extract_itrf(filepath: Path) -> str:
-    ext = filepath.suffix
-
-    if ext == ".qi":
-        header = _qfit_file_header(filepath)
-        results = re.finditer(r"itrf\d{2,4}", header)
-        itrfs = list({result.group() for result in results})
+# NOTE: See the docstring of `_infer_qfit_itrf` for more context about these
+# hard-coded ITRF date-ranges.
+# TODO: a typed-dict might be better here.
```

Comment:
These hard-coded ranges and the code to lookup an ITRF based on a date come straight from `valkyrie`, with some minor modifications.